app/core/middleware/logger.py

```diff
@@ -1,59 +1,125 @@
+        super().__init__(app)
 
     # https://github.com/tiangolo/fastapi/issues/394#issuecomment-927272627
-    # async def set_body(self, request: Request):
-    #     receive_ = await request._receive()
+    async def set_body(self, request: Request):
+        receive_ = await request._receive()
 
-    #     async def receive() -> Message:
-    #         return receive_
+        async def receive() -> Message:
+            return receive_
 
-    #     request._receive = receive
+        request._receive = receive
 
     async def dispatch(self, request, call_next):
-            start_time = time.time()
+        if request.url.path.startswith("/api/v1/health"):
+            response = await self._execute_request(call_next, request)
+            return
+        
+        request_id: str = str(uuid4())
+        logging_dict = {
+            "X-API-REQUEST-ID": request_id  # X-API-REQUEST-ID maps each request-response to a unique ID
+        }
 
-            # await self.set_body(request)
-            # req_body = await request.body()
-            response = await call_next(request)
+        await self.set_body(request)
+        response, response_dict = await self._create_response_log(
+            call_next,
+            request,
+            request_id
+        )
+        request_dict = await self._create_request_log(request)
+        logging_dict["req"] = request_dict
+        logging_dict["res"] = response_dict
 
-            # Consuming FastAPI response and grabbing body here
-            # resp_body = [section async for section in response.__dict__['body_iterator']]
-            # Repairing FastAPI response
-            # response.__setattr__('body_iterator', aiwrap(resp_body))
+        self._logger.info(logging_dict)
+        if self._file_logger is not None:
+            resp_body = [section async for section in response.__dict__["body_iterator"]]
+            response.__setattr__("body_iterator", AsyncIteratorWrapper(resp_body))
 
-            # Formatting response body for logging
-            # try:
-            #     resp_body = json.loads(resp_body[0].decode())
-            # except:
-            #     resp_body = str(resp_body)
+            try:
+                resp_body = json.loads(resp_body[0])
+            except:
+                resp_body = str(resp_body)
 
-            request.app.logger.info(
+            response_dict["body"] = resp_body
+            logging_dict["res"] = response_dict
+
+            self._file_logger.debug(logging_dict)
+
+        return response
+    
+    async def _create_request_log(self, request: Request) -> str:
+
+        path = request.url.path
+        if request.query_params:
+            path += f"?{request.query_params}"
+
+        request_log_dict = {
+            "method": request.method,
+            "endpoint": request.url.path,
+            "user": request.user.onyen if hasattr(request.user, "onyen") else None,
+        }
+
+        try:
+            body = await request.json()
```

Comment:
unless you've cloned something here I'm pretty sure you can't consume the request body inside the logger or it's not going to be readable in the actual handler? have you tested?

Reply:
There's a copy of the request body in the `_json` field on the request. `request.json()` just pulls that copy. I've tested it, the actual endpoint in the router has the full request body and can use it.

Reply:
You can check out the logs in the grader-api in the `eduhelx-test` namespace to see how the logs look